driver.go

```diff
@@ -14,62 +15,77 @@ type FireboltDriver struct {
 	lastUsedDsn  string
 }
 
+const engineStatusRunning = "Running"
+
 // Open parses the dsn string, and if correct tries to establish a connection
 func (d FireboltDriver) Open(dsn string) (driver.Conn, error) {
 	infolog.Println("Opening firebolt driver")
 
 	if d.lastUsedDsn != dsn || d.lastUsedDsn == "" {
 
 		d.lastUsedDsn = "" //nolint
-		infolog.Println("constructing new client")
-		// parsing dsn string to get configuration settings
+		infolog.Println("Creating a new client")
 		settings, err := ParseDSNString(dsn)
 		if err != nil {
 			return nil, ConstructNestedError("error during parsing a dsn", err)
 		}
 
-		// authenticating and getting access token
-		infolog.Println("dsn parsed correctly, trying to authenticate")
-		d.client, err = Authenticate(settings.username, settings.password, GetHostNameURL())
-		if err != nil {
-			return nil, ConstructNestedError("error during authentication", err)
+		if err = validateSettings(settings); err != nil {
+			return nil, ConstructNestedError("invalid connection string", err)
 		}
 
-		// getting accountId, either default, or by specified accountName
-		var accountId string
-		if settings.accountName == "" {
-			infolog.Println("account name not specified, trying to get a default account id")
-			accountId, err = d.client.GetDefaultAccountId(context.TODO())
-		} else {
-			accountId, err = d.client.GetAccountIdByName(context.TODO(), settings.accountName)
+		// authenticating and getting access token
+		if d.client, err = Authenticate(settings.clientId, settings.clientSecret, GetHostNameURL()); err != nil {
+			return nil, ConstructNestedError("authentication error", err)
 		}
+
+		systemEngineURL, err := d.client.GetSystemEngineURL(context.TODO(), settings.accountName)
 		if err != nil {
-			return nil, ConstructNestedError("error during getting account id", err)
+			return nil, ConstructNestedError("error getting system engine url", err)
 		}
 
-		// getting engineUrl either by using engineName if available,
-		// if not using default engine for the database
-		if settings.engineName != "" {
-			if strings.Contains(settings.engineName, ".") {
-				d.engineUrl, err = makeCanonicalUrl(settings.engineName), nil
-			} else {
-				d.engineUrl, err = d.client.GetEngineUrlByName(context.TODO(), settings.engineName, accountId)
-			}
+		if len(settings.engine) > 0 {
+			infolog.Println("Connected to a system engine")
+			d.engineUrl = systemEngineURL
+			d.databaseName = settings.database
 		} else {
-			infolog.Println("engine name not set, trying to get a default engine")
-			d.engineUrl, err = d.client.GetEngineUrlByDatabase(context.TODO(), settings.database, accountId)
-		}
-		if err != nil {
-			return nil, ConstructNestedError("error during getting engine url", err)
+			engineUrl, status, dbName, err := d.client.GetEngineUrlStatusDBByName(context.TODO(), settings.engine, systemEngineURL)
```

Comment:
Shouldn't this be other way around? If settings.engine is supplied, try to connect to said engine?